src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/JsonShreddingMetricsFactory.java

```diff
@@ -0,0 +1,19 @@
+package io.stargate.sgv2.jsonapi.api.v1.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Singleton;
+
+/**
+ * Configures the {@link JsonShreddingMetricsReporter} instance that going to be injectable and used
+ * in the app.
+ */
+public class JsonShreddingMetricsFactory {
+  /** Replaces the CDI producer for JsonShreddingMetricsReporter built into Quarkus. */
+  @Singleton
```

Comment:
Probably cannot be singleton because we need new instance every time -- since it has state, `Timer.Sample`.